plugins/modules/cloud/lxd/lxd_project.py

```diff
@@ -0,0 +1,418 @@
+        type: path
+    trust_password:
+        description:
+          - The client trusted password.
+          - You need to set this password on the LXD server before
+            running this module using the following command.
+            lxc config set core.trust_password <some random password>
+            See U(https://www.stgraber.org/2016/04/18/lxd-api-direct-interaction/)
+          - If trust_password is set, this module send a request for
+            authentication before sending any requests.
+        required: false
+        type: str
+notes:
+  - Projects must have a unique name. If you attempt to create a project
+    with a name that already existed in the users namespace the module will
+    simply return as "unchanged".
+'''
+
+EXAMPLES = '''
+# An example for creating a project
+- hosts: localhost
+  connection: local
+  tasks:
+    - name: Create a project
+      community.general.lxd_project:
+        name: ansible-test-project
+        state: present
+        config: {}
+        description: my new project
+
+# An example for renaming a project
+- hosts: localhost
+  connection: local
+  tasks:
+    - name: Create a project
+      community.general.lxd_project:
+        name: ansible-test-project
+        new_name: ansible-test-project-new-name
+        state: present
+        config: {}
+        description: my new project
+'''
+
+RETURN = '''
+old_state:
+  description: The old state of the project
+  returned: success
+  type: str
+  sample: "absent"
+logs:
+  description: The logs of requests and responses.
+  returned: when ansible-playbook is invoked with -vvvv.
+  type: list
+  sample: "(too long to be placed here)"
+actions:
+  description: List of actions performed for the project.
+  returned: success
+  type: list
+  sample: '["create"]'
+'''
+from ansible_collections.community.general.plugins.module_utils.lxd import LXDClient, LXDClientException
+from ansible.module_utils.basic import AnsibleModule
+import os
+
+# ANSIBLE_LXD_DEFAULT_URL is a default value of the lxd endpoint
+ANSIBLE_LXD_DEFAULT_URL = 'unix:/var/lib/lxd/unix.socket'
+
+# PROJECTS_STATES is a list for states supported
+PROJECTS_STATES = [
+    'present', 'absent'
+]
+
+# CONFIG_PARAMS is a list of config attribute names.
+CONFIG_PARAMS = [
+    'config', 'description'
+]
+
+
+class LXDProjectManagement(object):
+    def __init__(self, module):
+        """Management of LXC projects via Ansible.
+
+        :param module: Processed Ansible Module.
+        :type module: ``object``
+        """
+        self.module = module
+        self.name = self.module.params['name']
+        self._build_config()
+        self.state = self.module.params['state']
+        self.new_name = self.module.params.get('new_name', None)
+
+        self.key_file = self.module.params.get('client_key')
+        if self.key_file is None:
+            self.key_file = '{0}/.config/lxc/client.key'.format(
+                os.environ['HOME'])
```

Comment:
Better use
```suggestion
            self.key_file = os.expanduser('~/.config/lxc/client.key'))
```
(same below)